src/operator-docs/utility/do.ts

```diff
@@ -1,6 +1,73 @@
 import { OperatorDoc } from '../operator.model';
 
 export const doOperator: OperatorDoc = {
-  'name': 'do',
-  'operatorType': 'utility'
+  name: 'do',
+  operatorType: 'utility',
+  signature:
+    'public do(nextOrObserver: Observer | function, error: function, complete: function): Observable',
```

Comment:
You could mark the optional parameters with a question mark:

public do(nextOrObserver?: Observer | function, error?: function, complete?: function): Observable

Reply:
I agree with you it will be better to have a special mark for optional parameters. My only concert that this is inconsistent with the rest of the document. I think you should make this as a suggestion for the whole documentation, then we can follow it as a rule. 
Right now, I don't think I should be the only one making this change.

Reply:
I've opened #168 which encompass this suggestion with others.